src/devSVG.cpp

```diff
@@ -36,16 +49,41 @@ class SVGDesc {
       pageno(0),
       standalone(standalone_),
       cc(gdtools::context_create()) {
-    file = fopen(R_ExpandFileName(filename.c_str()), "w");
-  }
 
-  bool ok() const {
-    return file != NULL;
+    if (filename == ":terminal:") {
+      stream = (std::ostream*) &(Rcpp::Rcout);
+      stream_type = 0;
+    } else if (filename == ":string:") {
+      stream = new std::ostringstream();
```

Comment:
I think we'd be better off avoiding explicit memory management, and instead using a `boost::shared_ptr()`.